### PR TITLE
Refactor demdownload

### PR DESCRIPTION
**Pull request by jlmaurer**
Refactor the DEM calculation

## Description

- Refactored demdownload to make the code clearer and easier to test
- renamed demdownload.py to dem.py 
- Added unit tests for dem functions
- Fixes bugs in the DEM download script

## Motivation and Context

- DEM functions functions were scattered over several different modules
- Lack of unit tests for the DEM functions
- This is one part of the line-of-sight PR that I'm breaking up

## How Has This Been Tested?

- Added a suite of unit tests to test the DEM module. 
- Updated the relative tolerance of the test_scenario_1 unit test

- test_dem.py and also added tests to test_interpolator.py

## Type of change

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x ] I have added an explanation of what this change does and why you'd like us to include them.
- [x ] I have written new tests for the core changes, as applicable.
- [x ] I have successfully ran tests with your changes locally.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/test_scenario_1.py [hunk 1]
```diff
@@ -68,5 +68,5 @@ def test_tropo_delay(tmp_path):
             SCENARIO_DIR, "hydro.envi"), userNDV=0.)
 
         # get the true delay from the weather model
-        assert np.allclose(wet, true_wet, equal_nan=True)
-        assert np.allclose(hydro, true_hydro, equal_nan=True)
+        assert np.allclose(wet, true_wet, equal_nan=True, rtol=1e-3)
+        assert np.allclose(hydro, true_hydro, equal_nan=True, rtol=1e-3)
```
Comment (dbekaert) [inline test/test_scenario_1.py:72]: nice

File: tools/RAiDER/interpolator.py [hunk 2]
```diff
@@ -100,3 +106,27 @@ def fillna3D(array, axis=-1):
...
+    nPixLat = demRaster.shape[0]
+    nPixLon = demRaster.shape[1]
+    xlats = np.linspace(minlat, maxlat, nPixLat)
+    xlons = np.linspace(minlon, maxlon, nPixLon)
+#    interpolator = RegularGridInterpolator(
+#        grid=(xlats, xlons), 
+#        values=demRaster,
+#        fill_value = np.nan
+#    )
+    interpolator = rgi(
```
Comment (dbekaert) [inline tools/RAiDER/interpolator.py:124]: might be better to track the projection information and use gdal warp with an interpolation scheme. Also see the tropo notebook on ISCE-docs page which has a GDAL interpolation command.

File: tools/RAiDER/interpolator.py [hunk 3]
```diff
@@ -100,3 +106,27 @@ def fillna3D(array, axis=-1):
     out = dfd.values.reshape(array.shape)
     
     return np.moveaxis(out,-1,axis)
+
+
+def interpolateDEM(demRaster, outLL, extent):
+    ''' Interpolate a DEM raster to a set of lat/lon query points '''
```
Comment (dbekaert) [inline tools/RAiDER/interpolator.py:112]: This function is assuming lon lat wgs84 projection of the dem. is this a regular grid also if you querry the datapoint on the model nodes that is not regular in lonlat?

File: tools/RAiDER/llreader.py [hunk 4]
```diff
@@ -35,23 +34,25 @@ def readLL(*args):
     if flag == 'files':
         # If they are files, open them
         lat, lon = args
-        lats, latproj, _ = gdal_open(lat, returnProj=True)
-        lons, lonproj, _ = gdal_open(lon, returnProj=True)
+        lats, llproj1, _ = gdal_open(lat, returnProj=True)
+        lons, llproj2, _ = gdal_open(lon, returnProj=True)
+        if llproj1 != llproj2:
+            raise ValueError('The project of the lat and lon files are not compatible')
```
Comment (dbekaert) [inline tools/RAiDER/llreader.py:40]: the project => the projection coordinate system?

**Comment by jlmaurer** [comment]
Closing this as it is now addressed by #137

**Pull request closed without merging**